homeassistant/components/light/lw12wifi.py

```diff
@@ -0,0 +1,166 @@
+            lw12_light: Instance of the LW12 controller.
+        """
+        self._light = lw12_light
+        self._name = name
+        self._state = None
+        self._effect = None
+        self._rgb_color = [255, 255, 255]
+        self._brightness = 255
+        # Setup feature list
+        self._supported_features = SUPPORT_BRIGHTNESS | SUPPORT_EFFECT \
+            | SUPPORT_COLOR | SUPPORT_TRANSITION
+
+    @property
+    def name(self):
+        """Return the display name of the controlled light."""
+        return self._name
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def hs_color(self):
+        """Read back the hue-saturation of the light."""
+        return color_util.color_RGB_to_hs(*self._rgb_color)
+
+    @property
+    def effect(self):
+        """Return current light effect."""
+        if self._effect is None:
+            return None
+        return self._effect.replace('_', ' ').title()
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._state
+
+    @property
+    def supported_features(self):
+        """Return a list of supported features."""
+        return self._supported_features
+
+    @property
+    def effect_list(self):
+        """Return a list of available effects.
+
+        Use the Enum element name for display.
+        """
+        import lw12
+        return [effect.name.replace('_', ' ').title()
+                for effect in lw12.LW12_EFFECT]
+
+    @property
+    def assumed_state(self) -> bool:
+        """Return True if unable to access real state of the entity."""
+        return True
+
+    @property
+    def shoud_poll(self) -> bool:
+        """Return False to not poll the state of this entity."""
+        return False
+
+    def turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        import lw12
+        self._light.light_on()
+        if ATTR_HS_COLOR in kwargs:
+            self._rgb_color = color_util.color_hs_to_RGB(
+                *kwargs.get(ATTR_HS_COLOR))
+            self._light.set_color(*self._rgb_color)
+            self._effect = None
+        if ATTR_BRIGHTNESS in kwargs:
+            self._brightness = kwargs.get(ATTR_BRIGHTNESS)
+            brightness = int(self._brightness / 255 * 100)
+            self._light.set_light_option(lw12.LW12_LIGHT.BRIGHTNESS,
+                                         brightness)
+        if ATTR_EFFECT in kwargs:
+            self._effect = kwargs.get(ATTR_EFFECT).replace(' ', '_').upper()
+            # Check if a known and supported effect was selected.
+            if self._effect not in [eff.name for eff in lw12.LW12_EFFECT]:
+                # Unknown effect was set, recover by disabling the effect
+                # mode and raise an error.
+                self._effect = None
+                raise HomeAssistantError("Unknown effect selected.")
```

Comment:
It's more common to log an error and return.

Reply:
As I am already importing logging, but not using it, this is a fair point.